include/Gaffer/MetadataAlgo.h

```diff
@@ -100,6 +100,16 @@ bool ancestorAffectedByChange( const Plug *plug, IECore::TypeId changedNodeTypeI
 /// Copies metadata from one target to another. The exclude pattern is used with StringAlgo::matchMultiple().
 void copy( const GraphComponent *from, GraphComponent *to, const StringAlgo::MatchPattern &exclude = "", bool persistentOnly = true, bool persistent = true );
 
+///
+enum Overwrite
+{
+	KeepExisting,
+	OverwriteExisting
+};
+
+/// Copy nodule and noodle color meta data from srcPlug to dstPlug
+void copyColors( const Gaffer::Plug *srcPlug , Gaffer::Plug *dstPlug, Overwrite overwrite );
```

Comment:
I'd value your thoughts on how this fits in with the existing `copy()` method above. I hate the abundance of arguments to that method, and was somewhat forced into adding a `name` argument in my BoxUI PR, and on top of that, it seems that the `overwrite` argument has a place there too. Then there's the "magic bool" arguments that  need commenting `/* persistent = */` at all the call sites if they're to make any sense.

I do like that the `enum Overwrite` approach avoids the "magic bool" problem, but it's something I've avoided so far in Gaffer, partly for the lame reason that I didn't want loads of little enums lying around, but also because a large proportion of the calls will be made from Python, where I think `copy( overwrite = True )` is genuinely better than `copy( overwrite = MetadataAlgo.Overwrite.KeepExisting )`.

While thinking about this with regard to `copy()` the other day, I can across [tagged_bool](https://github.com/akrzemi1/explicit/blob/master/doc/tagged_bool.md), which seemed kindof interesting. I was also wondering if it made any sense to instead use an enum/bitmask approach to combine all three bool arguments into one, but again that seems less readable than just using keyword arguments in Python.

What are your thoughts? I tend to think I'd be happy enough with bools if only we had keyword arguments in C++. But in the meantime I would like to keep the Gaffer API consistent, and also have `overwrite` be available in `copy()` too.